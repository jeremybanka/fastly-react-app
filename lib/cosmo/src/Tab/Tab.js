/* @flow */

import type { ComponentType } from "react";
import styled from "styled-components";
import css from "@styled-system/css";

type Props = {
  active: boolean,
  children: React.Node,
};

const Tab: ComponentType<Props> = styled("a")((props) =>
  css({
    display: "block",
    borderRadius: "100px",
    paddingY: "sm",
    paddingX: "md",
    textDecoration: "none",
    backgroundColor: props.active && "indigos.1",
    color: props.active ? "indigo" : "grays.4",
    fontWeight: "semibold",
    "&:hover, &:focus": {
      outline: "0",
      backgroundColor: !props.active && "grays.1",
    },
  })
);

/** @component */
export default Tab;
