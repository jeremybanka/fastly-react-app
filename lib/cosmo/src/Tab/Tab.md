Use inside `Box` to stack vertically, or `Flex` to stack horizontally

```jsx
<Tab active marginBottom='md'>Lights</Tab>
<Tab marginBottom='md'>Camera</Tab>
<Tab marginBottom='md'>Action</Tab>
```
