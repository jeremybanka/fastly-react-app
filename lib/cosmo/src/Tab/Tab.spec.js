/* @flow */

import React from "react";
import { render } from "enzyme";

import Tab from "./Tab";

describe("Tab test suite", () => {
  it("renders the stored snapshot", () => {
    const component = (
      <Tab active href="#asdf">
        Hi
      </Tab>
    );
    expect(render(component)).toMatchSnapshot();
  });
});
