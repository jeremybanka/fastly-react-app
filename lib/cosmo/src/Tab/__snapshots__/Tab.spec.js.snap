// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab test suite renders the stored snapshot 1`] = `
<a
  class="Box-sc-1v2etov-0 Tab___StyledBox-sc-1b9e60u-0 eQuihY"
  href="#asdf"
>
  <span
    class="Text-sc-6c1xd4-0 gJmlqU"
    color="indigo"
    font-weight="semibold"
  >
    Hi
  </span>
</a>
`;
